public/app/features/explore/CorrelationEditorModeBar.tsx

```diff
@@ -0,0 +1,156 @@
+import { css } from '@emotion/css';
+import React, { useEffect, useState } from 'react';
+import { Prompt } from 'react-router-dom';
+import { useBeforeUnload, usePrevious } from 'react-use';
+
+import { GrafanaTheme2, colorManipulator } from '@grafana/data';
+import { Button, HorizontalGroup, Icon, Tooltip, useStyles2 } from '@grafana/ui';
+import { ExploreItemState, useDispatch, useSelector } from 'app/types';
+
+import { CorrelationUnsavedChangesModal } from './CorrelationUnsavedChangesModal';
+import { removeCorrelationHelperData } from './state/explorePane';
+import { changeCorrelationEditorDetails, changeCorrelationEditorMode } from './state/main';
+import { runQueries, saveCurrentCorrelation } from './state/query';
+import { selectCorrelationDetails, selectCorrelationEditorMode } from './state/selectors';
+
+// we keep component rendered and hidden to avoid race conditions with the prompt
```

Comment:
I had done this because originally there was just the logic if it's in editor mode, it shows, otherwise it hides. With the various confirmation modals also in the editor mode bar component, it got a little more complicated. I'll take a look at this and have it so the editorMode isn't change to false until after it's confirmed.

Reply:
Does the comment still apply after your recent changes? It seems we don't flip the visibility anymore.

Reply:
Nope! removed!